### ISSUE TITLE
The ca_extend::extend_ca_cert plan fails on PE 2021.0 with regen_primary_cert=true

### ISSUE THREAD

**Issue by Sharpie** [puppetlabs/ca_extend#12]
## Describe the Bug

When run against a PE 2021.0 standard install, the `ca_extend::extend_ca_cert` plan fails with `regen_primary_cert=true` as `puppet infrastructure configure` can't regenerate a missing primary certificate. For added fun, `infra configure` exits with a success code, so the plan _appears_ to work, but `puppet agent -t` still fails.

## Expected Behavior

`bolt plan run ca_extend::extend_ca_cert regen_primary_cert=true` results in `puppet agent -t` working when run against an expired PE 2021.0 installation.

## Steps to Reproduce
- Acquire a virtual machine running CentOS 7.
- Set the system clock back in time 5 years:
```
timedatectl set-ntp false
timedatectl set-time "$(date --date='5 years ago' +'%Y-%m-%d %H:%M:%S')"
```
- Install PE 2018.1.18. Run `puppet agent -t` twice to verify everything works.
- Upgrade to PE 2019.8.5
- Upgrade to PE 2021.0
- Run `puppet agent -t` to verify everything still works.
- Re-enable NTP with `timedatectl set-ntp true`. `puppet agent -t` should now fail as all certificates have expired.
- Use m0dular/crl_truncate to clear out the expired CRL (this is rare to encounter on a real-world install as some revocation activity usually occurs to regularly advance the CRL expiration date)
- Run the `ca_extend::extend_ca_cert` plan with `regen_primary_cert=true`
- Run `puppet agent -t`

## Environment
 - Version: puppetlabs-ca_extend 2.1.0
 - Platform: Any PE primary platform

## Additional Context

Output from `puppet infrastructure configure` shows an early failure when trying to run `puppetserver ca generate` to recreate the primary certificate:

```
Error: Failed to initialize SSL: The private key is missing from '/etc/puppetlabs/puppet/ssl/private_keys/temporal-test.platform9.puppet.net.pem'
Error: Run `puppet agent -t`
Warning: Error connecting to temporal-test.platform9.puppet.net on 8081 at route https://temporal-test.platform9.puppet.net:8081/pdb/query/v4, error message received was 'Request to https://temporal-test.platform9.puppet.net:8081/pdb/query/v4 failed after 0.003 seconds: The private key is missing from '/etc/puppetlabs/puppet/ssl/private_keys/temporal-test.platform9.puppet.net.pem''. Failing over to the next PuppetDB server_url in the 'server_urls' list
Error: Failed to initialize SSL: The private key is missing from '/etc/puppetlabs/puppet/ssl/private_keys/temporal-test.platform9.puppet.net.pem'
Error: Run `puppet agent -t`
2021-03-17T18:34:01.467+00:00 - [Notice]: Compiled catalog for temporal-test.platform9.puppet.net in environment enterprise in 5.81 seconds
2021-03-17T18:34:10.343+00:00 - [Notice]: /Stage[main]/Pe_install::Prepare::Certificates/Exec[generate node cert]/returns: Error:
2021-03-17T18:34:10.343+00:00 - [Notice]: /Stage[main]/Pe_install::Prepare::Certificates/Exec[generate node cert]/returns: Existing file at '/etc/puppetlabs/puppet/ssl/private_keys/temporal-test.platform9.puppet.net.pem'
2021-03-17T18:34:10.343+00:00 - [Notice]: /Stage[main]/Pe_install::Prepare::Certificates/Exec[generate node cert]/returns: Existing file at '/etc/puppetlabs/puppetserver/ca/signed/temporal-test.platform9.puppet.net.pem'
2021-03-17T18:34:10.343+00:00 - [Notice]: /Stage[main]/Pe_install::Prepare::Certificates/Exec[generate node cert]/returns: Please delete these files if you really want to generate a new cert for temporal-test.platform9.puppet.net.
2021-03-17T18:34:10.346+00:00 - [Error]: '/opt/puppetlabs/bin/puppetserver ca generate --certname temporal-test.platform9.puppet.net --ca-client --subject-alt-names 'puppet'' returned 1 instead of one of [0]
2021-03-17T18:34:10.350+00:00 - [Error]: /Stage[main]/Pe_install::Prepare::Certificates/Exec[generate node cert]/returns: change from 'notrun' to ['0'] failed: '/opt/puppetlabs/bin/puppetserver ca generate --certname temporal-test.platform9.puppet.net --ca-client --subject-alt-names 'puppet'' returned 1 instead of one of [0]
2021-03-17T18:34:10.352+00:00 - [Notice]: /Stage[main]/Puppet_enterprise::Pxp_agent::Service/Service[pxp-agent]: Dependency Exec[generate node cert] has failures: true
...
```

**Comment by Sharpie** [comment]
One solution to this issue may be to `case` on `/opt/puppetlabs/server/pe_version` and use `puppet infrastructure run --force regenerate_master_certificate` instead of `find -delete` + `puppet infrastructure configure` if the PE version is newer than 2019.

**Comment by m0dular** [comment]
We should be able to just add `puppet config print cadir` to the `find`, right?  I can put up a PR for that.

**Comment by MartyEwings** [comment]
SUP-2447

**Comment by m0dular** [comment]
Opened https://github.com/puppetlabs/ca_extend/pull/28

**Comment by m0dular** [comment]
Fixed in #28
